File: pset1/cash/cash.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

int main(void)
{
    // get change form user input
    int change = 0;
    do
    {
        change = get_int("Change owed: ");
    }
    while (change < 0); // Ensure valid integer input and positive value

    // get coin values and no. of coins
    int quarter = 25;
    int dime = 10;
    int nickel = 5;
    int penny = 1;
    int coins = 0;

    while (change != 0)
    {
        // check if change is larger than each coin (starting from the largest denomination)
        if (change >= quarter)
        {
            change -= quarter;
        }
        else if (change >= dime)
        {
            change -= dime;
        }
        else if (change >= nickel)
        {
            change -= nickel;
        }
        else if (change >= penny)
        {
            change -= penny;
        }

        coins++;
    }

    printf("%i\n", coins);
}

File: pset6/sentimental-cash/cash.py
from cs50 import get_float

# Get change
change = None
while (not isinstance(change, float) or change < 0):
    try:
        change = float(get_float("Change: "))
    except:
        pass

# Multiply change by 100 to remove float
change = int(change * 100)

# Get coin values
quarter, dime, nickel, penny = 25, 10, 5, 1

# Coin counter
coins = 0

# Calculate change w/ number of coins
while change > 0:
    # check if change is larger than each coin (starting from the largest denomination)
    if change >= quarter:  # 0.25
        change -= quarter
    elif change >= dime:  # 0.10
        change -= dime
    elif change >= nickel:  # 0.05
        change -= nickel
    elif change >= penny:  # 0.01
        change -= penny

    # increment coin
    coins += 1

# print coins
print(coins)
